pkg/controller/atlasproject/custom_roles.go

```diff
@@ -220,7 +220,7 @@ func syncCustomRolesStatus(ctx *workflow.Context, desiredCustomRoles []customrol
 			statuses = append(statuses, stat)
 
 			if stat.Status == status.CustomRoleStatusFailed {
-				err = errors.Join(err, fmt.Errorf(stat.Error))
+				err = errors.Join(err, fmt.Errorf("%w", stat.Error))
```

Comment:
stat.Error is a `string`

```suggestion
				err = errors.Join(err, fmt.Errorf("%s", stat.Error))
```